.gitignore

```diff
@@ -13,3 +13,7 @@ webapp/dist
 webapp/node_modules
 !.vscode/README.md
 !sdkconfig.defaults*
+webapp/package-lock.json
```

Comment:
`package-lock.json` should be committed because it guarantees reproducible installs. Removing it allows dependency versions to drift, which can cause inconsistent builds and hard-to-trace bugs.